client/src/main/java/org/eclipse/hono/client/impl/AmqpAdapterClientEventSenderImpl.java

```diff
@@ -0,0 +1,103 @@
+ */
+public class AmqpAdapterClientEventSenderImpl extends EventSenderImpl implements EventSender, TraceableEventSender {
+
+    AmqpAdapterClientEventSenderImpl(final HonoConnection con, final ProtonSender sender, final String tenantId,
+            final String targetAddress) {
+        super(con, sender, tenantId, targetAddress);
+    }
+
+    /**
+     * Creates a new sender for publishing events to Hono's AMQP adapter.
+     *
+     * @param con The connection to the Hono server.
+     * @param tenantId The tenant that the events will be published for.
+     * @param remoteCloseHook The handler to invoke when the link is closed by the peer (may be {@code null}). The
+     *            sender's target address is provided as an argument to the handler.
+     * @return A future indicating the outcome.
+     * @throws NullPointerException if con or tenantId is {@code null}.
+     */
+    public static Future<EventSender> createWithAnonymousLinkAddress(
+            final HonoConnection con,
+            final String tenantId,
+            final Handler<String> remoteCloseHook) {
+
+        Objects.requireNonNull(con);
+        Objects.requireNonNull(tenantId);
+
+        final String targetAddress = null; // anonymous relay
+        return con.createSender(targetAddress, ProtonQoS.AT_LEAST_ONCE, remoteCloseHook)
+                .map(sender -> new AmqpAdapterClientEventSenderImpl(con, sender, tenantId, targetAddress));
+    }
+
+    @Override
+    public Future<ProtonDelivery> send(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties, final SpanContext context) {
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(payload);
+        Objects.requireNonNull(contentType);
+
+        final Message msg = ProtonHelper.message();
+        msg.setAddress(getTo(deviceId));
+        MessageHelper.setPayload(msg, contentType, payload);
+        setApplicationProperties(msg, properties);
+        MessageHelper.addDeviceId(msg, deviceId);
+        return send(msg, context);
+    }
+
+    @Override
+    public Future<ProtonDelivery> send(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties) {
+        return send(deviceId, payload, contentType, properties, null);
+    }
+
+    /**
+     * Shadowing the static factory method that must not be used outside a protocol adapter.
+     *
+     * @param con not used.
+     * @param tenantId not used.
+     * @param remoteCloseHook not used.
+     * @return nothing.
+     * @throws UnsupportedOperationException always.
+     */
+    public static Future<DownstreamSender> create(
+            final HonoConnection con,
+            final String tenantId,
+            final Handler<String> remoteCloseHook) {
+        throw new UnsupportedOperationException("This method must not be used outside a protocol adapter");
```

Comment:
IMHO, shadowing the (unsupported) parent static factory method with what is now in `createWithAnonymousLinkAddress` would be a better idea (i.e. renaming `createWithAnonymousLinkAddress` to just `create`). WDYT?